src/main/kotlin/org/kamiblue/client/util/math/VectorUtils.kt

```diff
@@ -63,6 +64,19 @@ object VectorUtils {
         return IntRange((d1 - d2).floorToInt(), (d1 + d2).ceilToInt())
     }
 
+    fun Vec2f.toViewVec(): Vec3d {
+        val rotationRad = toRadians()
+        val yaw = -rotationRad.x - PI_FLOAT
+        val pitch = -rotationRad.y
+
+        val cosYaw = cos(yaw)
+        val sinYaw = sin(yaw)
+        val cosPitch = -cos(pitch)
+        val sinPitch = sin(pitch)
+
+        return Vec3d((sinYaw * cosPitch).toDouble(), sinPitch.toDouble(), (cosYaw * cosPitch).toDouble())
+    }
```

Comment:
Could you leave a few comments explaining how this math works?

Reply:
It is just basic trigonometry math tho

Reply:
i'm tired